test/solidity/Facets/EverclearFacet.t.sol

```diff
@@ -0,0 +1,1102 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.17;
+
+import { ILiFi } from "lifi/Interfaces/ILiFi.sol";
+import { LibSwap } from "lifi/Libraries/LibSwap.sol";
+import { TestBaseFacet } from "../utils/TestBaseFacet.sol";
+import { EverclearFacet } from "lifi/Facets/EverclearFacet.sol";
+import { LibAllowList } from "lifi/Libraries/LibAllowList.sol";
+import { IEverclearFeeAdapter } from "lifi/Interfaces/IEverclearFeeAdapter.sol";
+import { InvalidCallData, InvalidConfig, InvalidNonEVMReceiver, InvalidReceiver, NativeAssetNotSupported } from "lifi/Errors/GenericErrors.sol";
+
+// Stub EverclearFacet Contract
+contract TestEverclearFacet is EverclearFacet {
+    constructor(address _feeAdapter) EverclearFacet(_feeAdapter) {}
+
+    function addDex(address _dex) external {
+        LibAllowList.addAllowedContract(_dex);
+    }
+
+    function setFunctionApprovalBySignature(bytes4 _signature) external {
+        LibAllowList.addAllowedSelector(_signature);
+    }
+}
+
+contract EverclearFacetTest is TestBaseFacet {
+    EverclearFacet.EverclearData internal validEverclearData;
+    TestEverclearFacet internal everclearFacet;
+    IEverclearFeeAdapter internal constant FEE_ADAPTER =
+        IEverclearFeeAdapter(
+            address(0x2944F6fEF163365A382E9397b582bfbeB7C4F300)
+        );
+
+    uint256 internal signerPrivateKey;
+    address internal signerAddress;
+
+    // values defaultUSDCAmount and fee taken from quote data where totalFeeBps is 0.6509
+    // quote data from:
+    //   const quoteResp = await fetch(
+    //     `https://api.everclear.org/routes/quotes`,
+    //     {
+    //       method: 'POST',
+    //       headers: { 'Content-Type': 'application/json' },
+    //       body: JSON.stringify({
+    //         "origin": "42161",
+    //         "destinations": [
+    //           "10"
+    //         ],
+    //         "inputAsset": "0xaf88d065e77c8cC2239327C5EDb3A432268e5831",
+    //         "amount": "100000000",
+    //         "to": "0x2b2c52B1b63c4BfC7F1A310a1734641D8e34De62"
+    //       })
+    //     }
+    //   )
+    uint256 internal usdCAmountToSend = 99934901; // its defaultUSDCAmount - fee (100000000 - 65099)
```

Comment:
would be cleaner to just calculate it instead of hardcoding (and then having to explain it)